react/components/EditorContainer/Topbar/index.tsx

```diff
@@ -9,43 +12,94 @@ interface Props {
   changeMode: (mode?: StoreEditMode) => void
   mode?: StoreEditMode
   urlPath: string
+  onChangeUrlPath: (url: string) => void
   visible: boolean
 }
 
 const Topbar: React.FunctionComponent<Props> = ({
   changeMode,
   mode,
+  onChangeUrlPath,
   urlPath,
   visible,
-}) => (
-  <div
-    className={
-      visible ? 'ph5 f6 h-3em w-100 flex justify-between items-center' : 'dn'
+}) => {
+  const editor = useEditorContext()
+  const [urlInputDisabled, setUrlInputDisabled] = useState(false)
+  const [url, setUrl] = useState(urlPath)
+
+  useEffect(
+    () => {
+      setUrl(urlPath)
+    },
+    [urlPath]
+  )
+
+  const onEnter = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+    callback: (...args: any[]) => any
+  ) => event.key === 'Enter' && callback()
+
+  const handleNavigateToUrl = () => {
+    if (url === urlPath) {
+      return
+    }
+    setUrlInputDisabled(true)
+    onChangeUrlPath(url)
+  }
+
+  const handleUrlKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
:pick: Even though the `void` return type is implicit, it may be a good idea to explicitly set it to make sure you do intend to have a void function and it is not forgetting to return something.